src/test/java/homework4/CollectionsIntoMapTest.java

```diff
@@ -0,0 +1,45 @@
+package homework4;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Map;
+
+import static homework4.CollectionsIntoMap.convertCollectionsIntoMap;
+import static homework4.CollectionsIntoMap.convertCollectionsIntoMapException;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+class CollectionsIntoMapTest {
+
+    ArrayList<Integer> firstFiveDimensionalList = new ArrayList<>(Arrays.asList(1, 2, 3, 4, 5));
+    ArrayList<Integer> secondFiveDimensionalList = new ArrayList<>(Arrays.asList(6, 7, 8, 9, 10));
+    ArrayList<Integer> secondFourDimensionalList = new ArrayList<>(Arrays.asList(6, 7, 8, 9));
+    ArrayList<Integer> secondSixDimensionalList = new ArrayList<>(Arrays.asList(6, 7, 8, 9, 10, 11));
+
+    @Test
+    void twoCollectionsSameSize() {
+        assertEquals(Map.of(1, 6, 2, 7, 3, 8, 4, 9, 5, 10), convertCollectionsIntoMap(firstFiveDimensionalList, secondFiveDimensionalList));
+    }
+
+    @Test
+    void firstCollectionSizeGreaterThanSecond() {
```

Comment:
Does this test pass on your machine? for me it fails because "null" is not the same as just a null Object that you put inside of your method:
<img width="1684" alt="image" src="https://user-images.githubusercontent.com/90613721/182462526-ff459fbc-fbf0-4bc1-8e47-36e9f36c1a31.png">


Also, you can't use Map.of(5, null) to pass a null, because its arguments are non-nullable. to fix this issue you need something like this:
<img width="830" alt="image" src="https://user-images.githubusercontent.com/90613721/182462591-39073acd-3994-469c-afd3-683d5cc747ba.png">

Reply:
Thanks, done